### ISSUE TITLE
Add ability to specify custom attributes for ng-model element

### ISSUE THREAD

**Issue by kentcdodds** [formly-js/angular-formly#378]
I would like to keep the "magic" of angular formly that reduces the watchers to the best of its abilities (using `required="true"` rather than `ng-required="to.required"` when angular-formly knows that it will never change for example).

However, I would also like to allow users to specify attributes that they want to go directly onto the `ng-model` element with the `'attribute-name': 'attribute-value'` type notation. Something like this:

```javascript
{
  type: 'input',
  ngModelElAttrs: {
    'my-custom-directive': 'to.myCustomDirectiveValue'
  },
  expressionProperties: {
    'templateOptions.myCustomDirectiveValue': function() {
      return 'something that changes or whatever';
    }
  }
}
```

This would result in something like:

```html
<input ng-model="model[options.key]" my-custom-directive="to.myCustomDirectiveValue" />
```

There would be no magic involved at all. It would simply add the attributes by name and value to the template before it's compiled and that's it.

Any thoughts on the API or the concept? I think this would make things much easier to integrate with existing directive libraries.

Note, this would NOT be a breaking change as it is additive :-)

**Comment by MacKentoch** [comment]
As it is just additive, I think it is 100% OK.

I'm sure I may find application later in my form generator.

**Comment by kentcdodds** [comment]
Awesome. I think that I'm going to implement this one. I think it's a good idea and much simpler than the existing `ngModelAttrs`.

**Comment by ckniffen** [comment]
How is this different than just setting ngModelAttrs with the different options?

```
ngModelAttrs: {
  dateDisplayFormat: {
    attribute: 'date-format',
    bound: 'something-date-format'
  }
}
```

**Comment by kentcdodds** [comment]
The existing `ngModelAttrs` is a highly complicated API because it's highly optimized. This would just make things more straightforward.

**Comment by sparty02** [comment]
I like this, but I thnk a better name would be helpful.  The reference to ngModel (while I get it), is initially confusing as it implies that you are defining something for the model object, when you are really defining attributes for the HTML (generally the input element) that formly is generating for you.  This was/is confusing for ngModelAttrs to.  Just thinking out loud; can't think of a better name at the moment.

**Comment by kentcdodds** [comment]
@sparty02, while your concern is valid, I think that the most straightforward name would be something like: `ngModelElementAttributes`, but that's way too wordy. If we abbreviate the stuff that makes sense, we can do: `ngModelElAttrs` and that's much better. I can't think of any other name that would be more strightforward.

**Comment by MacKentoch** [comment]
:+1:
